### PR TITLE
add geonames db of italian cap-city-province and full match when generating full address

### PR DESCRIPTION
**Pull request by FrancYescO**
also improved format for building_number_formats and secondary_address_formats

Maybe the code can be improved but ... 🤷

Fix (at least for italy) https://github.com/joke2k/faker/issues/1585

PS. i've used this snippet to "convert to dictionary" the geonames db
```
import csv
fulldict = {}
with open('IT.txt', newline = '') as games:                                                                                          
	game_reader = csv.reader(games, delimiter='\t')
	for game in game_reader:
		if game[1] not in fulldict:
			fulldict[game[1]] = [[game[2],game[6]]]
		else:
			fulldict[game[1]].append([game[2],game[6]])

print(fulldict)
```

### CODE REVIEW

**Comment by FrancYescO** [comment]
> faker/providers/phone_number/en_PH/__init__.py:17:121: E501 line too long (122 > 120 characters)

dafaq? en_PH was jus untouched..

File: faker/providers/address/it_IT/__init__.py [hunk 1]
```diff
@@ -511,11 +17616,19 @@ class Provider(AddressProvider):
...
-    secondary_address_formats = ("Appartamento ##", "Piano #")
+    address_formats = ("{{street_address}}\n{{postcode_city_province}}",)
+    secondary_address_formats = ("Appartamento @#", "Piano #")
+
+    def postcode_city_province(self) -> str:
+        cap = self.postcode()
+        rand_city_prov = self.random_element(self.cap_city_province[cap])  # type: List[str]
+        return cap + ", " + rand_city_prov[0] + " (" + rand_city_prov[1] + ")"
+
+    def city(self):
```
Comment (fcurella) [inline faker/providers/address/it_IT/__init__.py]: can you add a type hint here too?

```suggestion
    def city(self) -> str:
```

File: faker/providers/address/it_IT/__init__.py [hunk 2]
```diff
@@ -511,11 +17616,19 @@ class Provider(AddressProvider):
...
+        "{{street_name}}, {{building_number}} {{secondary_address}}",
     )
-    address_formats = ("{{street_address}}\n{{city}}, {{postcode}} {{state}} ({{state_abbr}})",)
-    secondary_address_formats = ("Appartamento ##", "Piano #")
+    address_formats = ("{{street_address}}\n{{postcode_city_province}}",)
+    secondary_address_formats = ("Appartamento @#", "Piano #")
+
+    def postcode_city_province(self) -> str:
+        cap = self.postcode()
+        rand_city_prov = self.random_element(self.cap_city_province[cap])  # type: List[str]
```
Comment (fcurella) [inline faker/providers/address/it_IT/__init__.py]: we fully support type hints, so you dont need the py2-compatible syntax
```suggestion
        rand_city_prov: List[str] = self.random_element(self.cap_city_province[cap])
```

**Review by fcurella — CHANGES_REQUESTED** [review]
Can you add some tests for `postcode_city_province` and `city`?

**Comment by fcurella** [comment]
@FrancYescO  Don't worry about that extra commit, we're going to squash them anyway. But please let's keep the comments family-friendly :)

**Comment by FrancYescO** [comment]
> Can you add some tests for `postcode_city_province` and `city`?

can you suggest a good place for the test case?

**Comment by fcurella** [comment]
@FrancYescO You should add a testcase in [tests/providers/test_address.py](https://github.com/joke2k/faker/blob/master/tests/providers/test_address.py)

**Comment by fcurella** [comment]
Perfetto! Grazie! <3

**Pull request merged**